.ci/azure-pipelines.yml

```diff
@@ -12,38 +12,129 @@ pr:
+
+    - script: 'npm ci'
+      displayName: 'Install Dependencies'
+
+    - script: 'npx cordova prepare'
+      displayName: 'Prepare Cordova'
+
+    - script: 'npx cordova build android --debug'
+      displayName: 'Build Debug'
+      condition: eq(variables['BuildConfiguration'], 'debug')
+
+    - script: 'npx cordova plugin rm cordova-plugin-chromecast'
+      displayName: 'Remove Play Services'
+      condition: eq(variables['ArtifactSuffix'], 'libre')
+
+    - script: 'npx cordova build android --release'
+      displayName: 'Build Release'
+      condition: eq(variables['BuildConfiguration'], 'release')
+
+    - task: CopyFiles@2
+      displayName: 'Copy APK'
+      inputs:
+        sourceFolder: 'platforms/android/app/build/outputs/apk'
+        contents: '**/*.apk'
+        targetFolder: '$(Build.ArtifactStagingDirectory)'
+        flattenFolders: true
+
+    - task: Bash@3
+      displayName: 'Rename APK'
+      inputs:
+        targetType: 'inline'
+        script: 'mv *.apk jellyfin-android-${ARTIFACTSUFFIX}-${TAG}.apk'
+        workingDirectory: '$(Build.ArtifactStagingDirectory)'
+
+    - task: PublishBuildArtifacts@1
+      displayName: 'Publish APK'
+      inputs:
+        pathToPublish: '$(Build.ArtifactStagingDirectory)'
+        artifactName: 'jellyfin-android-$(ArtifactSuffix)'
+        publishLocation: 'Container'
+
+  - job: Publish
+    displayName: 'Publish'
+
+    dependsOn: Build
+    condition: startsWith(variables['Build.SourceBranch'], 'refs/tags')
+
+    strategy:
+      matrix:
+        Debug:
+          BuildConfiguration: debug
+          ArtifactSuffix: debug
+        Release:
+          BuildConfiguration: release
+          ArtifactSuffix: release
+        Libre:
+          BuildConfiguration: release
+          ArtifactSuffix: libre
+
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    steps:
+    - script: 'echo "##vso[task.setvariable variable=TAG]$(git describe --tags)"'
+      displayName: 'Set Tag Variable'
+
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download APK'
+      inputs:
+        source: 'current'
+        artifact: 'jellyfin-android-$(ArtifactSuffix)'
+        path: '$(System.ArtifactsDirectory)'
+        runVersion: 'latest'
+
+    - task: AndroidSigning@3
+      displayName: 'Sign APK'
+      condition: eq(variables['BuildConfiguration'], 'release')
+      inputs:
+        apkFiles: '$(System.ArtifactsDirectory)/*.apk'
+        keystoreFile: keystore
+        keystorePassword: $(KeyStorePassword)
+
+    - task: GithubRelease@0
+      displayName: 'GitHub Upload'
+      inputs:
+        gitHubConnection: Jellyfin Release Download
+        repositoryName: jellyfin/jellyfin-android
+        assets: '$(System.ArtifactsDirectory)/*.apk'
+
+    - task: CopyFilesOverSSH@0
+      inputs:
+        sshEndpoint: repository
+        sourceFolder: '$(System.ArtifactsDirectory)'
+        contents: '**'
+        targetFolder: '/srv/jellyfin/repos/clients/android/$(TAG)'
```

Comment:
Is this pushing to build1 (a.k.a. repo1 again, once all this is done)? If so it should be `/srv/repository/releases/client/android/version/$(TAG)/` and, if possible, an SSH task run to update the symlink from `/srv/repository/releases/client/android/<blah>.apk` to that file.

Reply:
We also need to add the service connection for the repository or we can have the repo pull the file, which might be easier with the symlinks and the like.

Reply:
There's a couple ways to go about here, if it's OK we could keep using my `collect-apk.sh` script, which handles symlinking and such. So this would push into `/srv/repository/incoming/<project_name>_<tag_name>` then run `/srv/repository/collect-apk.sh /srv/repository/incoming/<project_name>_<tag_name>`. Still leaves some of my BASH in the picture, but might be cleaner than doing all that in the Azure CI.

Reply:
I was just going to drop the symlinks and expect users to select the newest version.